src/Main/ProgramSettings.h

```diff
@@ -34,15 +34,16 @@ struct ProgramSettings {
     bool version = false;
     bool help = false;
     std::vector<int> port;
-    int grpc_port = -1;
     int omp_thread_count = OMP_THREAD_COUNT;
     int event_thread_count = 2;
     std::string top_level_folder = "/";
     std::string starting_folder = ".";
     std::string host = "0.0.0.0";
     std::vector<std::string> files;
     std::string frontend_folder;
-    bool no_http = false;
+    bool no_http = false; // Deprecated
```

Comment:
What about explicitly marking this variable as deprecated so that we remember in the future to remove it?. 
```c++
[[deprecated]] bool no_http = false;
// or
bool no_http __attribute__((deprecated)) = false;
```
I prefer `[[deprecated]]` since it's more readable, and we already have [[maybe_unused]] in the codebase, so it's the same attribute style. Unfortunately, it will add a lot of clutter at compilation time, but it's a good reminder that improves code quality in the end since it forces us to keep our codebase clean.

It is also easy to remove the warning while we are developing. The following is enough:
```
cmake .. -DCMAKE_CXX_FLAGS="-Wno-deprecated-declarations"
```

Reply:
The renaming is a good idea; I will definitely do that. I would prefer not do do something that will add noise to compilation without a wider discussion -- I suggest making a separate issue to propose this.

Reply:
During the renaming, I realised that the `#ifndef` guard names in the headers weren't consistently renamed during the recent file shuffle. I'll see if I can fix these automatically (maybe I can add this to the code style script at some point).

Reply:
> During the renaming, I realised that the `#ifndef` guard names in the headers weren't consistently renamed during the recent file shuffle. I'll make a separate PR to fix these automatically using the style script.

Sorry, my bad. I also spotted a problem with `Hdf5Loader.h` since it looks like it isn't catching the header guard. Thanks for adding that to the script; it's very useful.

Reply:
I created issue #1023 for this.